src/test/java/com/salesforce/dynamodbv2/QueryTest.java

```diff
@@ -100,178 +106,134 @@ void queryWithKeyConditionsEq(TestArgument testArgument) {
 
     // test legacy calls (see {@code QueryMapper} for more on "legacy".
     /**
-     * Table has (hk, rkLow), (hk, rkMiddle), (hk, rkHigh); we ask for items that match hk and have rk > rkLow, so there
-     * should be 2 results: (hk, rkMiddle) and (hk, rkHigh).
+     * Table has (hk, RANGE_KEY_N_MIN), ..., (hk, RANGE_KEY_N_MAX); we ask for items that match hk and have
+     * rk > RANGE_KEY_N_MIN, so there should be RANGE_KEY_N_MAX - RANGE_KEY_N_MIN results:
+     * (hk, RANGE_KEY_N_MIN + 1), ..., (hk, RANGE_KEY_N_MAX).
      */
     @ParameterizedTest(name = "{arguments}")
     @ArgumentsSource(DefaultArgumentProvider.class)
     void queryWithKeyConditionsGtLow(TestArgument testArgument) {
         queryWithKeyConditionsComparisonOperatorInner(GT,
             testArgument,
-            RANGE_KEY_LOW_N_VALUE,
-            ImmutableSet.of(RANGE_KEY_MIDDLE_N_VALUE, RANGE_KEY_HIGH_N_VALUE));
+            String.valueOf(RANGE_KEY_N_MIN),
+            IntStream.rangeClosed(RANGE_KEY_N_MIN + 1, RANGE_KEY_N_MAX)
+                .mapToObj(String::valueOf)
+                .collect(Collectors.toSet()));
     }
 
     /**
-     * Table has (hk, rkLow), (hk, rkMiddle), (hk, rkHigh); we ask for items that match hk and have rk > rkMiddle, so
-     * there should be 1 result: (hk, rkHigh).
+     * Table has (hk, RANGE_KEY_N_MIN), ..., (hk, RANGE_KEY_N_MAX); we ask for items that match hk and have
+     * rk > RANGE_KEY_N_MIN + 1, so there should be RANGE_KEY_N_MAX - (RANGE_KEY_N_MIN + 1) results:
+     * (hk, RANGE_KEY_N_MIN + 2), ..., (hk, RANGE_KEY_N_MAX).
      */
     @ParameterizedTest(name = "{arguments}")
     @ArgumentsSource(DefaultArgumentProvider.class)
-    void queryWithKeyConditionsGtMiddle(TestArgument testArgument) {
+    void queryWithKeyConditionsGtAlmostAsLow(TestArgument testArgument) {
         queryWithKeyConditionsComparisonOperatorInner(GT,
             testArgument,
-            RANGE_KEY_MIDDLE_N_VALUE,
-            ImmutableSet.of(RANGE_KEY_HIGH_N_VALUE));
+            String.valueOf(RANGE_KEY_N_MIN + 1),
+            IntStream.rangeClosed(RANGE_KEY_N_MIN + 2, RANGE_KEY_N_MAX)
+                .mapToObj(String::valueOf)
+                .collect(Collectors.toSet()));
     }
 
     /**
-     * Table has (hk, rkLow), (hk, rkMiddle), (hk, rkHigh); we ask for items that match hk and have rk > rkHigh, so
-     * there should be 0 results.
-     */
-    @ParameterizedTest(name = "{arguments}")
```

Comment:
used to be 3 tests per operand (comparing with -1, comparing with 0, and comparing with -1—the three possible range keys), instead we now just have two per operand: compare with `RANGE_KEY_N_MIN` and compare with `RANGE_KEY_N_MIN + 1`